### PR TITLE
🤑 Buy sell amm

### PR DESCRIPTION
**Pull request by WRadoslaw**
#5018 
#5184

### CODE REVIEW

File: packages/atlas/src/joystream-lib/extrinsics.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ type PublicExtrinsic<TxFunction, ReturnValue> = TxFunction extends (...a: infer
   ? (...a: [...U, ExtrinsicStatusCallbackFn | undefined]) => Promise<ReturnValue>
   : never
 
-const PERMILLS_PER_PERCENTAGE = 10
+const PERMILLS_PER_PERCENTAGE = 10_000
```
Comment (thesan) [inline packages/atlas/src/joystream-lib/extrinsics.ts]: Wow so permill does mean "per million" and not the [`‰` permil](https://en.wikipedia.org/wiki/Per_mille). That's really a bad abbreviation choice on substrate!

File: packages/atlas/src/components/_crt/BuyMarketTokenModal/steps/BuyMarketTokenForm.tsx [hunk 2]
```diff
@@ -13,62 +14,68 @@ import { DetailsContent } from '@/components/_nft/NftTile'
...
+  setPrimaryButtonProps,
+  onSubmit,
+  pricePerUnit,
+}: BuyMarketTokenFormProps) => {
+  const { control, watch, handleSubmit, formState } = useForm<{ tokens: number | null }>()
   const { accountBalance } = useSubscribeAccountBalance()
-  const tokens = watch('tokens')
-  const { pricePerUnit, tokensOnSale, userBalance, title } = getTokenDetails(tokenId)
+  const tokens = watch('tokens') || 0
+  const { fullFee } = useFee('purchaseTokenOnMarketTx', ['1', '1', String(tokens ?? 0), '1000000'])
```
Comment (thesan) [inline packages/atlas/src/components/_crt/BuyMarketTokenModal/steps/BuyMarketTokenForm.tsx]: Did you mean to leave these dummy data ?

**Review by thesan — APPROVED** [review]
LGTM! I'm completely lost with the calculations though!

File: packages/atlas/src/components/_crt/BuyMarketTokenModal/steps/BuyMarketTokenForm.tsx (hunk 2 above)
Comment (WRadoslaw) [reply packages/atlas/src/components/_crt/BuyMarketTokenModal/steps/BuyMarketTokenForm.tsx]: Yeah, it doesnt really impact fees and calculating it here as well feels like a waste

**Pull request merged**